### PR TITLE
fix: Build and publish latest images automatically

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -42,4 +45,7 @@ jobs:
           DOCKERIO_ORG: ${{ secrets.DOCKERIO_ORG }}
         run: |
           tag=$(basename $GITHUB_REF)
+          if [ $tag = "master" ]; then
```
Comment (whynowy) [inline .github/workflows/release.yml:48]: Need clarification on `latest` tag, we define it as latest master build, or latest stable build? I had discussion with @VaibhavPage before, he thinks it should be the latter.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ on:
   push:
     tags:
       - 'v*'
+    branches:
+      - master
+      - fix-latest
```
Comment (whynowy) [inline .github/workflows/release.yml:9]: rm

File: .github/workflows/release.yml (hunk 1 above)
Comment (alexec) [reply .github/workflows/release.yml:48]: 1. It's currently neither of those. It is 3 months old.
2. This is not consistent with either workflows or CD. Both of which are the tip of master.
3. That isn't it's intent: https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375

File: .github/workflows/release.yml (hunk 2 above)
Comment (alexec) [reply .github/workflows/release.yml]: done

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**